Library/Homebrew/utils/formatter.rb

```diff
@@ -0,0 +1,57 @@
+require "utils/tty"
+
+module Formatter
+  module_function
+
+  def arrow(string, color: nil)
+    if color
+      "#{Tty.send(color)}==>#{Tty.reset} #{string}"
+    else
+      "==> #{string}"
+    end
+  end
+
+  def headline(string, color: nil)
+    arrow("#{Tty.bold}#{string}#{Tty.reset}", color: color)
+  end
+
+  def success(string)
+    colorize_text_before_colon(string, :green)
+  end
+
+  def warning(string)
+    colorize_text_before_colon(string, :yellow)
+  end
+
+  def error(string)
+    colorize_text_before_colon(string, :red)
+  end
+
+  def url(string)
+    descenders_no_underline("#{Tty.blue.underline}#{string}#{Tty.reset}")
+  end
+
+  def colorize_text_before_colon(string, color)
+    label, message = string.scan(/(.*?:)?(.*)/m)[0]
+
+    if label.nil?
+      "#{Tty.send(color)}#{message}#{Tty.reset}"
+    else
+      "#{Tty.send(color).underline}#{descenders_no_underline(label)}#{Tty.reset}#{message}"
+    end
+  end
+
+  def descenders_no_underline(string)
+    colon_reached = false
+
+    string.chars.map { |letter|
+      unless colon_reached
```

Comment:
Was this left in by accident? Doesn't the first line of `colorize_text_before_colon` take care of this?